classlib/src/main/java/org/teavm/classlib/java/util/stream/TCollectors.java

```diff
@@ -69,4 +73,46 @@ private TCollectors() {
         return TCollector.of(StringBuilder::new, accumulator, combiner,
                 sb -> sb.insert(0, prefix).append(suffix).toString());
     }
+
+    public static <E, K, V> TCollector<E, ?, Map<K, V>> toMap(Function<? super E, ? extends K> keyMapper,
+            Function<? super E, ? extends V> valueMapper) {
+        return TCollector.of(HashMap::new,
+                (map, el) -> {
+                    K k = keyMapper.apply(el);
+                    V newV = TObjects.requireNonNull(valueMapper.apply(el));
+                    V oldV = map.putIfAbsent(k, newV);
+                    if (oldV != null) {
+                        throw new IllegalStateException(
+                                "Key " + k + " corresponds to values " + oldV + " and " + newV);
+                    }
+                },
+                (m1, m2) -> {
+                    m2.forEach((k, v) -> {
```

Comment:
You better use `for` statement over map entries over `forEach`, since `forEach` produces extra overhead.